uc-crux-llvm/src/UCCrux/LLVM/Setup.hs

```diff
@@ -261,34 +215,48 @@ generate sym mts ftRepr selector (ConstrainedShape shape) =
           <$> annotatedLLVMPtr sym ?ptrWidth ftRepr selector
       (Shape.ShapePtr _constraints (Shape.ShapeAllocated n), FTPtrRepr _ptdTo) ->
         Shape.ShapePtr
-          <$> (SymValue <$> malloc sym ftRepr selector n)
+          <$> (SymValue <$> malloc sym ftRepr selector (fromIntegral n))
           <*> pure (Shape.ShapeAllocated n)
       (Shape.ShapePtr _constraints (Shape.ShapeInitialized vec), FTPtrRepr ptPtdTo) ->
         do
           let num = Seq.length vec
-          ptr <- malloc sym ftRepr selector num
-          let ftPtdTo = asFullType mts ptPtdTo
-          size <- sizeBv proxy sym ftPtdTo 1
+          Some numRepr <-
+            case NatRepr.mkNatRepr (fromIntegral num) of
```

Comment:
It is slightly safer to use `toEnum` here since it fixes both source and target types.  The `fromIntegral` can have silent wraparound effects because it's polymorphic in the target type (same concern for all the uses of `fromIntegral`in this code).  That doesn't apply here ATM, but just sharing from my bag of general Haskell practices.

Reply:
See #876